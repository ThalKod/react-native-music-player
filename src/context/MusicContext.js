import React from "react";

const MusicContext = React.createContext({
  selected: {},
  list: [],
  changeSelected: () => {}
});

export const MusicProvider = MusicContext.Provider;
export const MusicConsumer = MusicContext.Consumer;

export default MusicContext;
