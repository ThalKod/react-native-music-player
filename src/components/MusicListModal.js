import React, { useContext } from "react";
import {
  StyleSheet,
  Modal,
  Text,
  TouchableOpacity,
  View,
    ScrollView
} from "react-native";
import PropTypes from "prop-types";
import MaterialIcons from "react-native-vector-icons/MaterialIcons";
import MusicContext from "../context/MusicContext";
import { moderateScale } from "react-native-size-matters";


import MusicSingle from "../components/MusicSingle";

const MusicListModal = ({ isModalVisible, closeModal }) => {
  const { list } = useContext(MusicContext);
  const renderMusic = () => {
    return list.map((music) => {
      return <MusicSingle id={music.id} key={music.id} title={music.title} artist={music.author} imageUrl={music.cover} fileName={music.fileName}/>
    });
  };

  return (
      <Modal
          animationType="slide"
          transparent={true}
          visible={isModalVisible}
      >
        <View style={styles.container}>
          <View style={styles.content}>
            <View style={styles.header}>
              <TouchableOpacity style={styles.closeButton} onPress={closeModal}>
                <MaterialIcons name="close" size={moderateScale(26)} color="black"/>
              </TouchableOpacity>
              <Text style={styles.headerTitle}>My Music</Text>
            </View>
            <View
                style={{
                  borderBottomColor: 'black',
                  borderBottomWidth: StyleSheet.hairlineWidth,
                }}
            />
            <ScrollView>
              {list.length > 0 ? renderMusic() : (
                  <Text>No Music on the device</Text>
              )}
            </ScrollView>
          </View>
        </View>
      </Modal>
  )
};

MusicListModal.propTypes = {
  isModalVisible: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  content: {
    width: "100%",
    height: "90%",
    padding: "5%",
    backgroundColor: "white",
    borderTopLeftRadius: moderateScale(15),
    borderTopRightRadius: moderateScale(15),
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginBottom: moderateScale(15)
  },
  closeButton: {
    flex: 1
  },
  headerTitle: {
    flex: 2,
    fontSize: moderateScale(24),
    fontFamily: "Avenir-Roman"
  }
});

export default MusicListModal;

