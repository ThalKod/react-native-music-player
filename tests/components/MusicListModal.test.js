import React from "react";
import { create } from 'react-test-renderer';
import MusicListModal from "../../src/components/MusicListModal";
import { TouchableOpacity } from "react-native";
import { musicData } from "../__mock__/index";
import MusicContext from "../../src/context/MusicContext";


describe("MusicListModal", () => {
  let component = undefined;
  let closeModal = jest.fn();

  beforeEach(() => {
    const context = {
      list: musicData,
    };
    component = create(
        <MusicContext.Provider value={context}>
          <MusicListModal isModalVisible={true} closeModal={closeModal}/>
        </MusicContext.Provider>
    );
  });

  test("Should correctly render the component", () => {
      expect(component).toMatchSnapshot();
  });

  test("Should close the modal on close button pressed", () => {
    const root = component.root;
    const close = root.findAllByType(TouchableOpacity);

    close[0].props.onPress();
    expect(closeModal).toBeCalled();
  })
});
