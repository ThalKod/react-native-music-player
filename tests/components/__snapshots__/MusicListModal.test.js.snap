// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicListModal Should correctly render the component 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderTopLeftRadius": 23.57142857142857,
          "borderTopRightRadius": 23.57142857142857,
          "height": "90%",
          "padding": "5%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 23.57142857142857,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 40.85714285714286,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óóç
          </Text>
        </View>
        <Text
          style={
            Object {
              "flex": 2,
              "fontFamily": "Avenir-Roman",
              "fontSize": 37.714285714285715,
            }
          }
        >
          My Music
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "black",
            "borderBottomWidth": 0.5,
          }
        }
      />
      <RCTScrollView>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "height": 117.85714285714286,
                "justifyContent": "space-between",
                "marginTop": 23.57142857142857,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": 31.428571428571427,
                  "width": "25%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/images/default_music_cover.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 15.714285714285714,
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2,
                  "justifyContent": "space-around",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#353840",
                    "fontFamily": "Avenir-Black",
                    "fontSize": 28.285714285714285,
                  }
                }
              >
                Test music title
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#353840",
                      "fontFamily": "Avenir-Black",
                      "fontSize": 28.285714285714285,
                    },
                    Object {
                      "fontFamily": "Avenir-Roman",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                Unknown Artist
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "height": 117.85714285714286,
                "justifyContent": "space-between",
                "marginTop": 23.57142857142857,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": 31.428571428571427,
                  "width": "25%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/images/default_music_cover.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 15.714285714285714,
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2,
                  "justifyContent": "space-around",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#353840",
                    "fontFamily": "Avenir-Black",
                    "fontSize": 28.285714285714285,
                  }
                }
              >
                Picasso Test
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#353840",
                      "fontFamily": "Avenir-Black",
                      "fontSize": 28.285714285714285,
                    },
                    Object {
                      "fontFamily": "Avenir-Roman",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                Unknown Artist
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "height": 117.85714285714286,
                "justifyContent": "space-between",
                "marginTop": 23.57142857142857,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "height": "100%",
                  "marginRight": 31.428571428571427,
                  "width": "25%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/images/default_music_cover.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 15.714285714285714,
                    "flex": 1,
                    "height": null,
                    "resizeMode": "cover",
                    "width": null,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 2,
                  "justifyContent": "space-around",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#353840",
                    "fontFamily": "Avenir-Black",
                    "fontSize": 28.285714285714285,
                  }
                }
              >
                Test music title 2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#353840",
                      "fontFamily": "Avenir-Black",
                      "fontSize": 28.285714285714285,
                    },
                    Object {
                      "fontFamily": "Avenir-Roman",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                Unknown Artist
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
